src/traceViewer/web/ui/sourceTab.tsx

```diff
@@ -38,65 +40,36 @@ export const SourceTab: React.FunctionComponent<{
     return { fileName, lineNumber };
   }, [actionEntry]);
 
-  const content = useAsyncMemo<string>(async () => {
-    if (location.fileName)
-      return window.readFile(location.fileName);
-    return location.value || '';
-  }, [location.fileName, location.value], '');
-
-  const [editor, setEditor] = React.useState<{
-    editor: MonacoEditor.IStandaloneCodeEditor,
-    element: HTMLElement,
-    decorations: string[]
-  } | undefined>();
-  const [measure, ref] = useMeasure<HTMLDivElement>();
-
-  React.useLayoutEffect(() => {
-    if (!ref.current)
-      return;
-    if (editor && editor.element === ref.current)
-      return;
-
-    const standalone = monaco.editor.create(ref.current, {
-      value: '',
-      language: 'javascript',
-      readOnly: true
-    });
-    standalone.layout();
-    setEditor({ editor: standalone, element: ref.current, decorations: [] });
-  }, [ref, editor]);
-
-  React.useLayoutEffect(() => {
-    if (editor) {
-      editor.editor.setValue(content);
-      editor.decorations = decorateLine(editor.editor, location.lineNumber, editor.decorations);
+  const content = useAsyncMemo<string[]>(async () => {
+    const value = location.fileName ? await window.readFile(location.fileName) : location.value
+    const result = [];
+    let continuation: any;
+    for (const line of (value || '').split('\n')) {
+      const highlighted = highlightjs.highlight('javascript', line, true, continuation);
+      continuation = highlighted.top;
+      result.push(highlighted.value);
     }
-  }, [content, editor, location.lineNumber]);
-  React.useLayoutEffect(() => {
-    if (editor)
-      editor.decorations = decorateLine(editor.editor, location.lineNumber, editor.decorations);
-  }, [location.lineNumber, editor]);
+    return result;
+  }, [location.fileName, location.value], []);
 
+  const ref = React.createRef<HTMLDivElement>();
   React.useLayoutEffect(() => {
-    if (editor)
-      editor.editor.layout();
-  }, [editor, measure.width, measure.height]);
-
-  return <div ref={ref} style={{ flex: 'auto', minWidth: '0', minHeight: '0' }}></div>;
-};
+    if (ref.current)
+      ref.current.scrollIntoView({ block: 'center', inline: 'nearest' });
+  }, [content, location.lineNumber, ref]);
 
-function decorateLine(editor: MonacoEditor.IStandaloneCodeEditor, lineNumber: number | undefined, decorations: string[]): string[] {
-  if (lineNumber !== undefined) {
-    const result = editor.deltaDecorations(decorations, [{
-      range: new monaco.Range(lineNumber, 1, lineNumber, 1),
-      options: {
-        isWholeLine: true,
-        className: 'monaco-execution-line'
-      }
-    }]);
-    editor.revealLineInCenterIfOutsideViewport(lineNumber, 1);
-    return result;
-  } else {
-    return editor.deltaDecorations(decorations, []);
+  return <div className='source-tab'>{
+    content.map((markup, index) => {
+      const isHighlighted = (index + 1) === location.lineNumber;
```

Comment:
I got really confused here by isHighlighted, thinking it was related to highlight.js

Reply:
Renamed.